templates/prompts/gap-analysis.html

```diff
@@ -0,0 +1,114 @@
+2.  **`EXISTING_TEST_SUMMARY`:** This document describes the *actual* coverage of the current test suite, organized by the same categories.
+
+Your task is to act as an auditor. You must meticulously compare the `EXISTING_TEST_SUMMARY` against the `FEATURE_SPEC_SUMMARY` to determine if a "minimum satisfactory level of test coverage" has been met, based *explicitly* on the definition provided below.
+
+### **Critical Error Handling Rule**
+
+**Before all other instructions, apply this rule:** If either the `FEATURE_SPEC_SUMMARY` or the `EXISTING_TEST_SUMMARY` is empty, nonsensical, or clearly not the structured summary it is supposed to be (e.g., it contains an error message, is just random text), your **entire** response must be *only* the following string, followed by a brief explanation:
+`RESPONSE FAILED: <short explanation of the issue>`
+
+**Examples:**
+
+  * `RESPONSE FAILED: The FEATURE_SPEC_SUMMARY input was empty.`
+  * `RESPONSE FAILED: The EXISTING_TEST_SUMMARY input was empty.`
+  * `RESPONSE FAILED: The provided inputs are not valid summaries and cannot be compared.`
+
+Do not output *any* other text or formatting if this rule is triggered.
+
+-----
+
+### Definition of Minimum Satisfactory Coverage (If Successful)
+
+1.  **Test feature existence:** This is typically done with surface-level tests like `idlharness.js` for APIs or `parsing-testcommon.js` for CSS. These tests don't verify actual behavior.
+2.  **Test common use cases:** Use the feature in a realistic and straightforward way and verify the expected behavior.
+3.  **Test likely error scenarios:** Test realistic error scenarios. For example, out-of-bounds inputs, network errors, or the user rejecting a permission prompt.
+4.  **Test invalidation:** Cached results often need to be invalidated when the source of truth changes. This kind of test is common for CSS features, but can make sense for other features too. Often called "dynamic" when an initial state is updated by script.
+5.  **Test integration with other features:** If the feature integrates with other features in some meaningful way, test that the combination of the two features behaves as expected.
+
+-----
+
+### Required Output Structure
+
+Your response must be a formal coverage analysis structured as follows:
+
+1.  **Overall Verdict:** Start with a brief, direct conclusion (e.g., "VERDICT: Minimum satisfactory coverage has **NOT** been met.")
+2.  **Summary of Analysis:** In 1-2 sentences, explain the *primary reason* for your verdict (e.g., "While feature existence and common use cases are well-tested, the suite fails to cover any error scenarios or the critical invalidation behavior defined in the spec.")
+
+-----
+
+### Detailed Coverage Analysis
+
+For each category, compare the spec requirements to the test coverage and provide a verdict.
+
+### 1. Feature Existence
```

Comment:
Should this have `###`? That puts it at the same level as the section heading.

Reply:
You're right! I updated these subheaders to use `####`. 🙂